packages/jazz-browser/src/createBrowserContext.ts

```diff
@@ -163,50 +153,51 @@ export async function createJazzBrowserContext<Acc extends Account>(
   const { toggleNetwork, peersToLoadFrom, setNode, crypto } =
     await setupPeers(options);
 
-  AuthSecretStorage.migrate();
+  let unsubscribeAuthUpdate = () => {};
+
+  if (options.localOnly === "anonymous") {
+    const authSecretStorage = options.authSecretStorage;
+    const credentials = options.credentials ?? (await authSecretStorage.get());
 
-  const credentials = options.credentials ?? AuthSecretStorage.get();
+    authSecretStorage.emitUpdate(credentials);
+
+    function handleAuthUpdate(isAuthenticated: boolean) {
+      if (isAuthenticated) {
+        toggleNetwork(true);
+      } else {
+        toggleNetwork(false);
+      }
+    }
+
+    unsubscribeAuthUpdate = authSecretStorage.onUpdate(handleAuthUpdate);
+    handleAuthUpdate(authSecretStorage.isAuthenticated);
+  }
```

Comment:
Moved the auth based localOnly toggle inside here to make it easier to maintain.